waku/v2/protocol/lightpush/waku_lightpush.go

```diff
@@ -327,26 +328,42 @@ func (wakuLP *WakuLightPush) Publish(ctx context.Context, message *wpb.WakuMessa
 	req.Message = message
 	req.PubsubTopic = params.pubsubTopic
 
-	logger := message.Logger(wakuLP.log, params.pubsubTopic).With(logging.HostID("peerID", params.selectedPeer))
+	logger := message.Logger(wakuLP.log, params.pubsubTopic).With(zap.Stringers("peerIDs", params.selectedPeers))
 
 	logger.Debug("publishing message")
+	var wg sync.WaitGroup
+	var responses []*pb.PushResponse
+	for _, peerID := range params.selectedPeers {
+		wg.Add(1)
+		go func(id peer.ID) {
+			defer wg.Done()
+			response, err := wakuLP.request(ctx, req, params, id)
+			if err != nil {
+				logger.Error("could not publish message", zap.Error(err), zap.Stringer("peer", id))
+			}
+			responses = append(responses, response)
+		}(peerID)
+	}
+	wg.Wait()
+	var successCount int
+	errMsg := "lightpush error"
 
-	response, err := wakuLP.request(ctx, req, params)
-	if err != nil {
-		logger.Error("could not publish message", zap.Error(err))
-		return wpb.MessageHash{}, err
+	for _, response := range responses {
+		if response.GetIsSuccess() {
+			successCount++
+		} else {
+			if response.GetInfo() != "" {
+				errMsg += *response.Info
+			}
```

Comment:
Do you think it would make sense to indicate the peerId that generated the error?